lib/src/lifecycle.rs

```diff
@@ -2,8 +2,20 @@
 /// environment that particular test phases or milestones have been reached.
 
 use serde_json::{json, Value};
+use serde::Serialize;
 use crate::internal;
 
+#[derive(Serialize, Debug)]
+struct AntithesisSetupData<'a, 'b> {
+    status: &'a str,
+    details: &'b Value,
+}
+
+#[derive(Serialize, Debug)]
+struct SetupCompleteData<'a> {
+    antithesis_setup: AntithesisSetupData<'a, 'a>,
+}
+
```

Comment:
Maybe make those structs own the underlying data and remove the lifetime bounds?
This would make it easier to de-serialize in the future.